ops/model.py

```diff
@@ -3596,3 +3619,56 @@ def _ensure_loaded(self):
         self._notice = self._container.get_notice(self.id)
         assert self._notice.type == self.type
         assert self._notice.key == self.key
+
+
+@dataclasses.dataclass(frozen=True)
+class CloudCredential:
+    """CloudCredential contains a cloud credential possibly with secrets redacted.
```

Comment:
This reads oddly to me. Maybe split off the second bit? Or maybe you can think of a different way to word this?

> An object containing credentials for a Juju cloud.
>
> Note that secrets may be redacted.

Do we know *when* the secrets would be redacted, or in what way? I think if I was trying to use this I would like to know that, like it's always with AWS, or always with specific Juju versions, or something, and it's redacted by not being in `attributes` and being listed in `redacted` instead.